lib/src/chart/line_chart/line_chart_painter.dart

```diff
@@ -297,19 +298,26 @@ class LineChartPainter extends AxisChartPainter<LineChartData>
       }
 
       /// For drawing the indicator line
-      final bottom = Offset(touchedSpot.dx, getTopOffsetDrawSize() + chartViewSize.height);
-      final top = Offset(getPixelX(spot.x, chartViewSize), getTopOffsetDrawSize());
-
-      /// Draw to top or to the touchedSpot
-      final lineEnd =
-          data.lineTouchData.fullHeightTouchLine ? top : touchedSpot + Offset(0, dotHeight / 2);
+      final lineStartY =
+          min(data.maxY, max(data.minY, data.lineTouchData.getTouchLineStart(barData, index)));
+      final lineEndY =
+          min(data.maxY, max(data.minY, data.lineTouchData.getTouchLineEnd(barData, index)));
+      final lineStart = Offset(touchedSpot.dx, getPixelY(lineStartY, chartViewSize));
+      var lineEnd = Offset(touchedSpot.dx, getPixelY(lineEndY, chartViewSize));
+      if (data.lineTouchData.touchLineEndAtDot) {
+        if (lineStart.dy < lineEnd.dy) {
```

Comment:
What if `lineStart` is greater or equals to `lineEnd`?
I think we need to throw an exception if `getTouchLineEnd` provided a value lower than `getTouchLineEnd`.